src/test/java/HmWrk16.java

```diff
@@ -0,0 +1,47 @@
+import net.bytebuddy.asm.Advice;
+import org.checkerframework.checker.units.qual.K;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+public class HmWrk16 {
+
+    @Test
+    public static void createNewPlaylist() {
+
+        WebDriver driver = new ChromeDriver();
+        try {
+            driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+            String url = "https://bbb.testpro.io/";
+            driver.get(url);
+
+            By usernameLocator = By.xpath("//*[@type='email']");
+            By passwordLocator = By.xpath("//*[@type='password']");
+            By loginButton = By.xpath("//*[@type='submit']");
+            By avatarIconLocator = By.xpath("//*[@alt = 'Avatar of Student']");
+            By NewPlaylistBtn = By.xpath("//i[@title='Create a new playlist']");
+            By createNewPlaylist = By.xpath("//*[contains(text(),'New Playlist')]");
+            By newPlaylistName = By.xpath("//*[@name='name']");
+            By playlistName = By.xpath("//*[contains(text(),'MySongs')]");
+
+            driver.findElement(usernameLocator).sendKeys("demo@class.com");
+            driver.findElement(passwordLocator).sendKeys("te$t$tudent");
+            driver.findElement(loginButton).click();
+            driver.findElement(NewPlaylistBtn).click();
+            driver.findElement(createNewPlaylist).click();
+            driver.findElement(newPlaylistName).sendKeys("MySongs");
+            driver.findElement(newPlaylistName).sendKeys(Keys.RETURN);
+            Assert.assertTrue(driver.findElement(playlistName).isDisplayed());
+        } finally {
+            driver.quit();
+        }
+    }
+}
+
```

Comment:
I have corrected the errors in the classes BaseTest and  HmWrk16

For example

```
import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.BeforeSuite;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Optional;
import org.testng.annotations.Parameters;

public class BaseTest {

  WebDriver driver;
  String url;

  @BeforeSuite
  public static void chromeConfigs() {
    if (System.getProperty("os.name").toLowerCase().contains("win")) {
      System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
    } else {
      System.setProperty("webdriver.chrome.driver", "Untitled/chromedriver");
    }


  }

  @DataProvider(name = "invalidCredentials")
  public static Object[][] getCredentials() {

    return new Object[][]{
        {"invalid@class.com", "invalidPass"},
        {"d@class.com", ""},
        {"", ""}
    };
  }

  @BeforeMethod
  @Parameters({"baseURL"})
  public void launchBrowser(@Optional String baseURL) {
      if (baseURL == null) {
          baseURL = "https://bbb.testpro.io";
      }
    ChromeOptions chromeOptions = new ChromeOptions();
    chromeOptions.addArguments("--disable-notifications");
    driver = new ChromeDriver(chromeOptions);
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
    driver.manage().window().maximize();
    url = baseURL;
    driver.get(url);

  }

  @AfterMethod
  public void tearDownBrowser() {
    driver.quit();
  }

  public void clickSubmitBtn() {
    WebElement submitButton = driver.findElement(By.cssSelector("[type='submit']"));
    submitButton.click();
  }

  public void provideEmail(String email) {
    WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
    emailField.click();
    emailField.sendKeys(email);
  }

  public void providePassword(String password) {
    WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
    passwordField.click();
    passwordField.sendKeys(password);

  }

  public void login() {
    provideEmail("demo@class.com");
    providePassword("te$t$tudent");
    clickSubmitBtn();
  }
}
```

```
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.testng.Assert;
import org.testng.annotations.Test;


public class HmWrk16 extends BaseTest {

  By NewPlaylistBtn = By.xpath("//i[@title='Create a new playlist']");
  By createNewPlaylist = By.xpath("//*[contains(text(),'New Playlist')]");
  By newPlaylistName = By.xpath("//*[@name='name']");
  By playlistName = By.xpath("//*[contains(text(),'MySongs')]");


  @Test
  public void createNewPlaylist(){

   login();

    driver.findElement(NewPlaylistBtn).click();
    driver.findElement(createNewPlaylist).click();
    driver.findElement(newPlaylistName).sendKeys("MySongs");
    driver.findElement(newPlaylistName).sendKeys(Keys.RETURN);


    Assert.assertTrue(driver.findElement(playlistName).isDisplayed());

  }
}
```
Please fix the remaining errors and try to run the test

Reply:
im so lost, can this be explained

Reply:
**Don't worry, I'm going to help you.**

**Step 1** 

There is the class **BaseTest**

All test classes inherit (directly or indirectly) from the BaseTest class, which provides everything you need to create your own tests. It is a simple approach to set up common initialization and cleanup in your tests. We commonly categorize it as a Testing pattern.

BaseTest class holds the methods to initialize and terminate the WebDriver object. Since all the Test classes will extend the BaseTest class, then the initialization is done implicitly.

In this class, we declared the driver. This is used by all test classes. Also, we wrote @BeforeMethod setup method and @AfterMethod. Again, all test classes use these methods.

In the setup method, we create a ChromeDriver and maximize the browser.

In the teardown method, I closed all the browsers with driver.quit(); code line.

**Step 2**

There is the class **HmWrk16**

We create a test class called HmWrk16 

In test classes (Page Object Model with JAVA):

-We need to instantiate the required Page classes
-Use the appropriate page methods
-Do the assertions

_BaseTest class contains all common functionalities and variables of test classes and all test classes extend this BaseTest class. This is one of the main features of Object Oriented Design (OOD) “Inheritance“._